### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by naveensrinivasan**
What type of PR is this?
Enable codeql analysis to capture any security issues in code.

/kind cleanup

What this PR does / why we need it:
Discover vulnerabilities across a codebase with CodeQL

Which issue(s) this PR fixes:
None

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go', 'python' ]
```
Comment (willscott) [inline .github/workflows/codeql-analysis.yml:21]: there's no python in this repo. is there a reason you test it?

Comment (naveensrinivasan) [reply .github/workflows/codeql-analysis.yml:21]: The bot built that by analyzing the code. I can remove python. :)

**Review by Stebalien — APPROVED** [review]
After removing python, this seems reasonable. It's maintained by GitHub so there shouldn't be any security issues with pulling the action in directly rather than forking.

**Comment by Stebalien** [comment]
Thanks!

**Pull request merged**